compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -1413,6 +1413,18 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                     ConstraintCategory::Boring,
                 );
                 let sig = self.normalize(sig, term_location);
+                // HACK(#114936): `WF(sig)` does not imply `WF(normalized(sig))`
+                // with built-in `Fn` implementations, since the impl may not be
+                // well-formed itself.
+                self.prove_predicates(
```

Comment:
nit: do we want to only emit those in case normalization actually changed the arg/ret ty